### PR TITLE
Add workflow linter python

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by ghost** [comment]
```
librz/bin/d/dll/convert_dumpbin_exports_to_sdb_txt.py
librz/bin/format/xnu/scripts/machtraps.py
librz/bin/format/xnu/scripts/build_mig_index.py
```
are python2 scripts, we should remove them or update them to python3

**Comment by ghost** [comment]
> ```
> librz/bin/d/dll/convert_dumpbin_exports_to_sdb_txt.py
> librz/bin/format/xnu/scripts/machtraps.py
> librz/bin/format/xnu/scripts/build_mig_index.py
> ```
> 
> are python2 scripts, we should remove them or update them to python3

It is why the job is failing.

**Comment by ret2libc** [comment]
> > ```
> > librz/bin/d/dll/convert_dumpbin_exports_to_sdb_txt.py
> > librz/bin/format/xnu/scripts/machtraps.py
> > librz/bin/format/xnu/scripts/build_mig_index.py
> > ```
> > 
> > 
> > are python2 scripts, we should remove them or update them to python3
> 
> It is why the job is failing.

Is it possible to exclude them from the check for now?

**Comment by ghost** [comment]
> > > ```
> > > librz/bin/d/dll/convert_dumpbin_exports_to_sdb_txt.py
> > > librz/bin/format/xnu/scripts/machtraps.py
> > > librz/bin/format/xnu/scripts/build_mig_index.py
> > > ```
> > > 
> > > 
> > > are python2 scripts, we should remove them or update them to python3
> > 
> > 
> > It is why the job is failing.
> 
> Is it possible to exclude them from the check for now?

Yes, but it is a bad idea (https://www.python.org/doc/sunset-python-2/). If you want I can update them to python3 right now.

PS: I believe Rizin should remove all unmaintained dependencies.

**Comment by ret2libc** [comment]
> > > > ```
> > > > librz/bin/d/dll/convert_dumpbin_exports_to_sdb_txt.py
> > > > librz/bin/format/xnu/scripts/machtraps.py
> > > > librz/bin/format/xnu/scripts/build_mig_index.py
> > > > ```
> > > > 
> > > > 
> > > > are python2 scripts, we should remove them or update them to python3
> > > 
> > > 
> > > It is why the job is failing.
> > 
> > 
> > Is it possible to exclude them from the check for now?
> 
> Yes, but it is a bad idea (https://www.python.org/doc/sunset-python-2/). If you want I can update them to python3 right now.

If you are up for it... :D I won't stop you. Though I'd like to even know if those files are used and when/where.

**Comment by ghost** [comment]
> > > > > ```
> > > > > librz/bin/d/dll/convert_dumpbin_exports_to_sdb_txt.py
> > > > > librz/bin/format/xnu/scripts/machtraps.py
> > > > > librz/bin/format/xnu/scripts/build_mig_index.py
> > > > > ```
> > > > > 
> > > > > 
> > > > > are python2 scripts, we should remove them or update them to python3
> > > > 
> > > > 
> > > > It is why the job is failing.
> > > 
> > > 
> > > Is it possible to exclude them from the check for now?
> > 
> > 
> > Yes, but it is a bad idea (https://www.python.org/doc/sunset-python-2/). If you want I can update them to python3 right now.
> 
> If you are up for it... :D I won't stop you. Though I'd like to even know if those files are used and when/where.

I ask on Mattermost.

**Comment by ghost** [comment]
I removed grep convert_dumpbin_exports_to_sdb_txt.py because it is:

* undocumented
* not used anywhere (`grep -RIi convert_dumpbin_exports_to_sdb_txt.py`)

**Comment by ghost** [comment]
Ready for review.

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -90,3 +90,29 @@ jobs:
 
     - name: Run prettier
       run: find . -name "*.js" | grep -v "subprojects" | xargs npx prettier  --print-width 120 --check
+
+  python:
```
Comment (ret2libc) [inline .github/workflows/linter.yml:95]: I think i missed this `changes` job. What is it?

Comment (ghost) [reply .github/workflows/linter.yml:95]: It detect which files has been modified and set some variables.

In this case `needs.changes.outputs.python`

Comment (ghost) [reply .github/workflows/linter.yml:95]: So I can create a conditional run for the python linter (only run if a python files has been modified)

**Review by ret2libc — APPROVED** [review]
Merge whenever you like!

**Pull request merged**